src/backend/mscoffobj.c

```diff
@@ -708,7 +725,8 @@ void MsCoffObj::term(const char *objfilename)
         return;
 #endif
 
-    build_syment_table();
+    bool bigobj = scnhdr_cnt > 0xffff;
```

Comment:
The problematic number is not the number of symbols, but the number of sections.